### PR TITLE
Feature/objectest

### PR DESCRIPTION
**Pull request by aswath1**
# PR Details
## Description
Launch file update to launch
vision_darknet_detect
vision_beyond_track
lidar_euclidean_cluster_detect
range_vision_fusion
imm_ukf_pda_track
naive_motion_predict

## Motivation and Context
Adding object detection and tracking stack.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -54,7 +62,11 @@ RUN ~/carma_ws/src/CARMAPlatform/docker/install.sh
 # Stage 3 - Finalize deployment
 # /////////////////////////////////////////////////////////////////////////////
 
+<<<<<<< HEAD
+FROM usdotfhwastol/autoware.ai:latest
+=======
 FROM usdotfhwastol/autoware.ai:3.7.0
+>>>>>>> develop
```
Comment (msmcconnell) [inline Dockerfile]: Looks like you commited a merge conflict. Please resolve

File: carmajava/launch/environment.launch [hunk 2]
```diff
@@ -33,6 +33,124 @@
...
+
+  <!-- vision_darknet_detect -->
+  <group>
+  <include file="$(find vision_darknet_detect)/launch/vision_yolo3_detect.launch">
+  <arg name="gpu_device_id" default="0"/>
+  <arg name="score_threshold" default="0.30"/>
+  <arg name="nms_threshold" default="0.30"/>
+
+  <arg name="network_definition_file" default="$(find vision_darknet_detect)/darknet/cfg/yolov3.cfg"/>
+  <arg name="pretrained_model_file" default="$(find vision_darknet_detect)/darknet/data/yolov3.weights"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Probably better to point both of these paths to /opt/carma/yolo/ Then the volume in the docker compose could just be /opt/carma/yolo:/opt/carma/yolo

File: carmajava/launch/environment.launch [hunk 3]
```diff
@@ -33,6 +33,124 @@
...
+  <arg name="names_file" default="$(find vision_darknet_detect)/darknet/cfg/coco.names"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: I guess this file should also be in /opt/carma/yolo

File: carmajava/launch/environment.launch [hunk 4]
```diff
@@ -33,6 +33,124 @@
...
+  <arg name="image_src" default="$(optenv CARMA_INTR_NS)/camera/image_raw"/>
+  </include>
+  </group>
+
+<!-- vision_beyond_track -->
+  <group>
+  <include file="$(find vision_beyond_track)/launch/vision_beyond_track.launch">
+  <arg name="camera_info_src" default="$(optenv CARMA_INTR_NS)/camera/camera_info"/>
+  <arg name="objects_topic_src" default="detection/image_detector/objects"/>
+  <arg name="camera_height_calibration_params_file" default="$(find vehicle_calibration_dir)/vision_beyond_track/calibration.yaml"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: There is no vehicle_calibration_dir package. You should add an argument to the environment.launch file called vehicle_calibration_dir and pass it in from carma_src.launch where it already exists. See localization.launch for an example.

File: carmajava/launch/environment.launch [hunk 5]
```diff
@@ -33,6 +33,124 @@
...
+  </include>
+  </group>
+  
+  <!-- lidar_euclidean_cluster_detect -->
+  <group>
+  <include file="$(find lidar_euclidean_cluster_detect)/launch/lidar_euclidean_cluster_detect.launch">
+  <arg name="points_node" default="$(optenv CARMA_LOCZ_NS)/ray_ground_filter/points_no_ground" /><!--CHANGE THIS TO READ WHETHER FROM VSCAN OR POINTS_RAW -->
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Does this comment still apply?

File: carmajava/launch/environment.launch [hunk 6]
```diff
@@ -33,6 +33,124 @@
...
+  <arg name="remove_ground" default="true" />
+  <arg name="downsample_cloud" default="false" /> <!-- Apply VoxelGrid Filter with the value given by "leaf_size"-->
+  <arg name="leaf_size" default="0.1" /><!-- Voxel Grid Filter leaf size-->
+  <arg name="cluster_size_min" default="20" /><!-- Minimum number of points to consider a cluster as valid-->
+  <arg name="cluster_size_max" default="100000" /><!-- Maximum number of points to allow inside a cluster-->
+  <arg name="sync" default="false" />
+  <arg name="use_diffnormals" default="false" />
+  <arg name="pose_estimation" default="true" />
+  <arg name="clip_min_height" default="-1.3" />
+  <arg name="clip_max_height" default="0.5" />
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Are these two arguments related to hardware positions? Should they be in vehicle calibration?

File: carmajava/launch/environment.launch [hunk 7]
```diff
@@ -33,6 +33,124 @@
...
+  <arg name="keep_lanes" default="false" />
+  <arg name="keep_lane_left_distance" default="5" />
+  <arg name="keep_lane_right_distance" default="5" />
+  <arg name="cluster_merge_threshold" default="1.5" />
+  <arg name="clustering_distance" default="0.75" />
+
+  <arg name="use_vector_map" default="false" />
+  <arg name="wayarea_gridmap_layer" default="wayarea" />
+
+  <arg name="output_frame" default="velodyne" />
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Would this be concat_velodyne on the truck? If so we need to move it to vehicle calibration.

File: carmajava/launch/environment.launch [hunk 8]
```diff
@@ -33,6 +33,124 @@
...
+
+  </include>
+  </group>
+
+  <!-- naive_motion_predict -->
+  <group>
+  <include file="$(find naive_motion_predict)/launch/naive_motion_predict.launch">
+  <arg name="interval_sec" default="0.1"/>
+  <arg name="num_prediction" default="10"/>
+  <arg name="sensor_height_calibration_params_file" default="$(find vehicle_calibration_dir)/naive_motion_predict/calibration.yaml"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Same note as above about calibration folder

File: carmajava/launch/environment.launch (hunk 6 above)
Comment (aswath1) [reply carmajava/launch/environment.launch]: Its not hardware, its remove points below and above z distance (m)

**Comment by aswath1** [comment]
https://github.com/usdot-fhwa-stol/CARMAPlatform/issues/432

File: carmajava/launch/environment.launch [hunk 9]
```diff
@@ -18,6 +18,8 @@
 	Launch file for launching the nodes in the CARMA environmental perception stack
 -->
 <launch>
+  <arg name="vehicle_calibration_dir" default="$(find carma)../../CARMAVehicleCalibration/development/vehicle" doc="The directory continaing vehicle calibration type parameters"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: You need to pass this argument in from the carma_src.launch file

File: carmajava/launch/environment.launch [hunk 10]
```diff
@@ -35,6 +37,17 @@
...
   <!-- Object Tracking and World Model -->
 
+  <!-- calibration_publisher -->
+  
+  <group>
+  <include file="$(find calibration_publisher)/launch/calibration_publisher.launch">
+  <arg name="register_lidar2camera_tf" default="True" />
+  <arg name="target_frame" default="velodyne" />
+  <arg name="camera_frame" default="left_optical" />
+  <arg name="calibration_file" default="$(find calibration_publisher)/config/lidar_camera_fl_calibration.yaml"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: You still have the find syntax here. Needs to use the calibration directory

File: carmajava/launch/environment.launch [hunk 11]
```diff
@@ -72,24 +86,16 @@
   <arg name="sync" default="false" />
   <arg name="use_diffnormals" default="false" />
   <arg name="pose_estimation" default="true" />
-  <arg name="clip_min_height" default="-1.3" />
-  <arg name="clip_max_height" default="0.5" />
-
+  <arg name="calibration_params_file" default="$(find vehicle_calibration_dir)/lidar_euclidean_cluster_detect/calibration.yaml"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: Same as above

File: carmajava/launch/environment.launch [hunk 12]
```diff
@@ -41,9 +41,14 @@
   
   <group>
   <include file="$(find calibration_publisher)/launch/calibration_publisher.launch">
-  <arg name="register_lidar2camera_tf" default="True" />
+  <arg name="register_lidar2camera_tf" default="true" />
+  <arg name="publish_extrinsic_mat" default="true"/>
+  <arg name="publish_camera_info" default="true"/>  
   <arg name="target_frame" default="velodyne" />
   <arg name="camera_frame" default="left_optical" />
+  <arg name="image_topic_src" default="/hardware_interface/camera/image_color"/>
```
Comment (msmcconnell) [inline carmajava/launch/environment.launch]: You should always use the optenv syntax instead of /hardware_interface

**Comment by msmcconnell** [comment]
Is the external object node for carma being launched?
